File: Operators/reinit_continuous.m
function [output1,output2] = reinit_continuous(varargin)
% Reinitialization for continuous problems.

mode = varargin{end};
switch mode
    case 'execute'
        Solution = varargin{1};
        Problem  = varargin{2};

        N = length(Solution);
        Lower = Problem.bound(1,:);
        Upper = Problem.bound(2,:);
        output1 = unifrnd(repmat(Lower,N,1),repmat(Upper,N,1));
        output2 = varargin{5};

    case 'parameter'
        % n/a

    case 'behavior'
        output1 = {'';'GS'}; % always perform global search
end

if ~exist('output1','var')
    output1 = [];
end
if ~exist('output2','var')
    output2 = [];
end
end

File: AutoOptLib for Python/components/reinit_continuous.py
"""Python translation of reinit_continuous."""

from __future__ import annotations

import numpy as np


def reinit_continuous(*args):
    mode = args[-1]
    if mode == "execute":
        solution = args[0]
        problem = args[1] if len(args) > 1 else None
        aux = args[3] if len(args) > 3 else None

        try:
            n = len(solution)
        except TypeError:
            n = int(getattr(problem, "N", 1))
        bound = getattr(problem, "bound")
        lower = np.asarray(bound[0], dtype=float)
        upper = np.asarray(bound[1], dtype=float)
        dec = lower + (upper - lower) * np.random.rand(n, lower.shape[-1])
        return dec, aux

    if mode == "parameter":
        return None, None

    if mode == "behavior":
        return ["", "GS"], None

    raise ValueError(f"Unsupported mode: {mode}")
